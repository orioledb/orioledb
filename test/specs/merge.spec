setup
{
	CREATE EXTENSION IF NOT EXISTS orioledb;
	CREATE TABLE IF NOT EXISTS o_merge (
		id int8 NOT NULL,
		PRIMARY KEY(id)
	) USING orioledb;
	TRUNCATE o_merge;
	INSERT INTO o_merge (SELECT i FROM generate_series(1, 1000) AS i);
	SET enable_seqscan = off;
}

teardown
{
	DROP TABLE o_merge;
}

session "s1"

step "s1_count" { SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge; }
step "s1_bcount" { SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t; }
step "s1_begin" { BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ; }
step "s1_commit" { COMMIT; }
step "s1_structure" { SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1); }
step "s1_delete500_begin" { DELETE FROM o_merge WHERE id BETWEEN 1 AND 500; }
step "s1_delete500_end" { DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000; }
step "s1_delete_80_perc" { DELETE FROM o_merge WHERE id % 5 <> 0; }
step "s1_evict" { SELECT orioledb_evict_pages('o_merge'::regclass, 1); }

session "s2"

step "s2_count" { SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge; }
step "s2_bcount" { SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t; }
step "s2_begin" { BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ; }
step "s2_commit" { COMMIT; }
step "s2_structure" { SELECT orioledb_tbl_structure('o_merge'::regclass, 'nue', 1); }
step "s2_evict" { SELECT orioledb_evict_pages('o_merge'::regclass, 1); }

permutation "s1_structure" "s1_delete500_begin" "s1_evict" "s1_count" "s1_structure" "s1_delete500_end" "s1_evict" "s1_count" "s1_structure"
permutation "s1_structure" "s1_delete_80_perc" "s1_evict" "s1_count" "s1_structure"

permutation "s1_count" "s1_delete500_begin" "s1_evict" "s1_count" "s1_delete500_end" "s1_count"
permutation "s1_count" "s1_delete_80_perc" "s1_evict" "s1_count"
permutation "s2_count" "s1_delete500_begin" "s1_evict" "s2_count" "s1_delete500_end" "s1_evict" "s2_count"
permutation "s2_count" "s1_delete_80_perc" "s1_evict" "s2_count"

permutation "s1_bcount" "s1_delete500_begin" "s1_evict" "s1_bcount" "s1_delete500_end" "s1_evict" "s1_bcount"
permutation "s1_bcount" "s1_delete_80_perc" "s1_evict" "s1_bcount"
permutation "s2_bcount" "s1_delete500_begin" "s1_evict" "s2_bcount" "s1_delete500_end" "s1_evict" "s2_bcount"
permutation "s2_bcount" "s1_delete_80_perc" "s1_evict" "s2_bcount"

permutation "s1_begin" "s2_begin" "s1_count" "s1_delete500_begin" "s1_count" "s1_delete500_end" "s1_count" "s1_commit" "s1_evict" "s1_count" "s1_structure" "s2_commit"
permutation "s1_begin" "s2_begin" "s1_count" "s1_delete_80_perc" "s1_count" "s1_commit" "s1_evict" "s1_count" "s1_structure" "s2_commit"
permutation "s1_begin" "s2_begin" "s2_count" "s1_delete500_begin" "s2_count" "s1_delete500_end" "s2_count" "s1_commit" "s2_evict" "s2_count" "s2_structure" "s2_commit"
permutation "s1_begin" "s2_begin" "s2_count" "s1_delete_80_perc" "s2_count" "s1_commit" "s2_evict" "s2_count" "s2_structure" "s2_commit"

permutation "s1_begin" "s2_begin" "s1_bcount" "s1_delete500_begin" "s1_bcount" "s1_delete500_end" "s1_bcount" "s1_commit" "s1_evict" "s1_bcount" "s1_structure" "s2_commit"
permutation "s1_begin" "s2_begin" "s1_bcount" "s1_delete_80_perc" "s1_bcount" "s1_commit" "s1_evict" "s1_bcount" "s1_structure" "s2_commit"
permutation "s1_begin" "s2_begin" "s2_bcount" "s1_delete500_begin" "s2_bcount" "s1_delete500_end" "s2_bcount" "s1_commit" "s2_evict" "s2_bcount" "s2_structure" "s2_commit"
permutation "s1_begin" "s2_begin" "s2_bcount" "s1_delete_80_perc" "s2_bcount" "s1_commit" "s2_evict" "s2_bcount" "s2_structure" "s2_commit"
