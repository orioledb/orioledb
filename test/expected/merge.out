Parsed test spec with 2 sessions

starting permutation: s1_structure s1_delete500_begin s1_evict s1_count s1_structure s1_delete500_end s1_evict s1_count s1_structure
step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                                                                                                                                                                                        
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('306')
    Item 2: offset = 544, downlink = 3, key = ('611')
    Item 3: offset = 560, downlink = 4, key = ('916')


(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  500
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                                                                                                                                  
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('611')
    Item 2: offset = 544, downlink = 3, key = ('916')


(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
    0
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1


(1 row)


starting permutation: s1_structure s1_delete_80_perc s1_evict s1_count s1_structure
step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                                                                                                                                                                                        
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('306')
    Item 2: offset = 544, downlink = 3, key = ('611')
    Item 3: offset = 560, downlink = 4, key = ('916')


(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  200
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                                                                            
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('611')


(1 row)


starting permutation: s1_count s1_delete500_begin s1_evict s1_count s1_delete500_end s1_count
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  500
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
    0
(1 row)


starting permutation: s1_count s1_delete_80_perc s1_evict s1_count
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  200
(1 row)


starting permutation: s2_count s1_delete500_begin s1_evict s2_count s1_delete500_end s1_evict s2_count
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  500
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
    0
(1 row)


starting permutation: s2_count s1_delete_80_perc s1_evict s2_count
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  200
(1 row)


starting permutation: s1_bcount s1_delete500_begin s1_evict s1_bcount s1_delete500_end s1_evict s1_bcount
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  500
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
    0
(1 row)


starting permutation: s1_bcount s1_delete_80_perc s1_evict s1_bcount
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  200
(1 row)


starting permutation: s2_bcount s1_delete500_begin s1_evict s2_bcount s1_delete500_end s1_evict s2_bcount
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  500
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
    0
(1 row)


starting permutation: s2_bcount s1_delete_80_perc s1_evict s2_bcount
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  200
(1 row)


starting permutation: s1_begin s2_begin s1_count s1_delete500_begin s1_count s1_delete500_end s1_count s1_commit s1_evict s1_count s1_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  500
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
    0
(1 row)

step s1_commit: COMMIT;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
    0
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1


(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s1_count s1_delete_80_perc s1_count s1_commit s1_evict s1_count s1_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  200
(1 row)

step s1_commit: COMMIT;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
  200
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                                                                            
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('611')


(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s2_count s1_delete500_begin s2_count s1_delete500_end s2_count s1_commit s2_evict s2_count s2_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_commit: COMMIT;
step s2_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s2_structure: SELECT orioledb_tbl_structure('o_merge'::regclass, 'nue', 1);
orioledb_tbl_structure                                                                                                                                                                                                                                                              
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1

Index toast: not loaded

(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s2_count s1_delete_80_perc s2_count s1_commit s2_evict s2_count s2_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s1_commit: COMMIT;
step s2_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_count: SET enable_seqscan = off; SELECT COUNT(*) FROM o_merge;
count
-----
 1000
(1 row)

step s2_structure: SELECT orioledb_tbl_structure('o_merge'::regclass, 'nue', 1);
orioledb_tbl_structure                                                                                                                                                                                                                                                                                                                    
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('611')

Index toast: not loaded

(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s1_bcount s1_delete500_begin s1_bcount s1_delete500_end s1_bcount s1_commit s1_evict s1_bcount s1_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  500
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
    0
(1 row)

step s1_commit: COMMIT;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
    0
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1


(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s1_bcount s1_delete_80_perc s1_bcount s1_commit s1_evict s1_bcount s1_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  200
(1 row)

step s1_commit: COMMIT;
step s1_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s1_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
  200
(1 row)

step s1_structure: SELECT orioledb_idx_structure('o_merge'::regclass, 'o_merge_pkey', 'nue', 1);
orioledb_idx_structure                                                                                                                                                                                                                                                                                            
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, clean
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('611')


(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s2_bcount s1_delete500_begin s2_bcount s1_delete500_end s2_bcount s1_commit s2_evict s2_bcount s2_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete500_begin: DELETE FROM o_merge WHERE id BETWEEN 1 AND 500;
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete500_end: DELETE FROM o_merge WHERE id BETWEEN 501 AND 1000;
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_commit: COMMIT;
step s2_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s2_structure: SELECT orioledb_tbl_structure('o_merge'::regclass, 'nue', 1);
orioledb_tbl_structure                                                                                                                                                                                                                                                              
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1

Index toast: not loaded

(1 row)

step s2_commit: COMMIT;

starting permutation: s1_begin s2_begin s2_bcount s1_delete_80_perc s2_bcount s1_commit s2_evict s2_bcount s2_structure s2_commit
step s1_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_begin: BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_delete_80_perc: DELETE FROM o_merge WHERE id % 5 <> 0;
step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s1_commit: COMMIT;
step s2_evict: SELECT orioledb_evict_pages('o_merge'::regclass, 1);
orioledb_evict_pages
--------------------
                    
(1 row)

step s2_bcount: SET enable_seqscan = off; SELECT COUNT(*) FROM ( SELECT * FROM o_merge ORDER BY id DESC) t;
count
-----
 1000
(1 row)

step s2_structure: SELECT orioledb_tbl_structure('o_merge'::regclass, 'nue', 1);
orioledb_tbl_structure                                                                                                                                                                                                                                                                                                                    
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Index o_merge_pkey contents
Page 0: level = 1, maxKeyLen = 8
state = free, datoid equal, relnode equal, ix_type = primary, dirty
    Leftmost, Rightmost
  Chunk 0: offset = 0, location = 512, hikey location = 64
    Item 0: offset = 520, downlink = 1
    Item 1: offset = 528, downlink = 2, key = ('611')

Index toast: not loaded

(1 row)

step s2_commit: COMMIT;
